prov/gni/src/gnix_mr_cache.c

```diff
@@ -81,20 +85,19 @@ typedef struct gnix_mr_cache_entry {
 } gnix_mr_cache_entry_t;
 
 /* forward declarations */
-int _gnix_mr_cache_init(
-		gnix_mr_cache_t      **cache,
-		gnix_mr_cache_attr_t *attr);
```

Comment:
This seems to be going in the opposite direction of being a general purpose memory registration cache if we intend to use this with other providers later.